libyul/optimiser/OptimiserStep.h

```diff
@@ -37,6 +37,7 @@ struct OptimiserStepContext
 	Dialect const& dialect;
 	NameDispenser& dispenser;
 	std::set<YulString> const& reservedIdentifiers;
+	std::optional<size_t> expectedExecutionsPerDeployment;
```

Comment:
Currently, the wasm code transform didn't have access to `expectedExecutionsPerDeployment`, so I made it `nullopt` and the `keccak` would get evaluated.

Similarly, also for `YulOptimizerTest`. Although, for this, we could just replace `nullopt` by `200`.

Reply:
For the fuzzer to test the keccak rule in this PR, the check on expected runs may either be disabled (current state i.e., in fuzzer tests rule takes effect always) or set to a value that passes that check i.e., a runs that provides gas savings.

Since the fuzzer relies on YulOptimizerTest, please either set the runs to `nullopt` or some value that is large enough for there to be gas savings.